NBISweden/sda-cli
# Enable downloading encrypted file from sda-download API

**Related issue(s) and PR(s)**  
This PR closes #393 

**Description**
When the flag `-pubkey` is set, the subcommand `sda-download` will downloaded the encrypted file (and reencrypted by the provided public key) from the `sda-download` service 

**How to test**
1. Start the docker setup by running the following command in the root folder
```bash
bash .github/integration/setup/setup.sh 
```
2. Create a crypt4gh key pair by e.g. 
```bash
./sda-cli createKey c4gh
```
3. Download the dummy encrypted file with the following command
```bash
./sda-cli sda-download  -pubkey c4gh.pub.pem -config testing/s3cmd-download.conf -dataset https://doi.example/ty009.sfrrss/600.45asasga -url http://localhost:8080 -outdir test-download main/subfolder/dummy_data.c4gh
```
Note that the `filepath` can be provided with or without the `.c4gh` extension

After success run, the encrypted file will be downloaded to `test-download/main/subfolder/dummy_data.c4gh`

4. Try to decrypt the downloaded file with your private key by 
```bash
./sda-cli decrypt -key c4gh.sec.pem test-download/main/subfolder/dummy_data.c4gh
```
